### PR TITLE
Add support for "sync with system" as a theme option

### LINKED ISSUE #2703: Support "sync with system" as a theme option
Several sites (Github here included) provide users with an option to use their OS/device level "dark mode" setting to determine the theme. It would be great for us to provide the same as a third option, including making it the default.

Note that the system theme can be detected in JS via `window.matchMedia('(prefers-color-scheme: dark)').matches`.

A listener can be registered like so to ensure an already loaded site switches when the OS changes: 
```
window.matchMedia('(prefers-color-scheme: dark)').addListener((e) => console.log(`dark mode: ${e.matches}`))
```

### PR DESCRIPTION
**Pull request by ghsolomon**
Fixes #2703 

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by ghsolomon** [comment]
See also https://github.com/xh/hoist-core/pull/236

**Review by amcclain — APPROVED** [review]
Looks great - @lbwexler leaving open for you to have a look / catch up with this change, but all good to go from my perspective.

File: appcontainer/ThemeModel.js [hunk 1]
```diff
@@ -32,10 +32,25 @@ export class ThemeModel extends HoistModel {
...
         classList.toggle('bp3-dark', value);
         this.darkTheme = value;
-        XH.setPref('xhTheme', value ? 'dark' : 'light');
+
+    }
+
+    @action
+    setThemePref(value) {
+        if (value === 'system') {
+            this.setDarkTheme(window.matchMedia('(prefers-color-scheme: dark)').matches);
```
Comment (amcclain) [inline appcontainer/ThemeModel.js:41]: FYI the `matchMedia` API seems well-established / supported as per https://caniuse.com/matchmedia, so I don't _think_ we need to defend against mobile or other wonky browsers throwing here.

File: appcontainer/ThemeModel.js [hunk 2]
```diff
@@ -32,10 +32,25 @@ export class ThemeModel extends HoistModel {
         classList.toggle('xh-dark', value);
         classList.toggle('bp3-dark', value);
         this.darkTheme = value;
-        XH.setPref('xhTheme', value ? 'dark' : 'light');
+
+    }
+
+    @action
+    setThemePref(value) {
+        if (value === 'system') {
```
Comment (lbwexler) [inline appcontainer/ThemeModel.js:40]: think a switch with three clauses would actually be clearer at this point.

Comment (lbwexler) [inline appcontainer/ThemeModel.js:40]: think a switch with three clauses would actually be clearer at this point.

File: core/XH.js [hunk 3]
```diff
@@ -324,9 +324,20 @@ class XHClass extends HoistBase {
         return this.acm.themeModel.toggleTheme();
     }
 
-    /** Enable/disable the dark theme directly (useful for custom app option controls). */
+    /**
+     * Enable/disable the dark theme directly (useful for custom app option controls).
+     * @param {boolean} value
+     */
     setDarkTheme(value) {
```
Comment (lbwexler) [inline core/XH.js:331]: Think we should deprecate or remove `setDarkTheme()` and make "setTheme()" (without the suffix) the main entry point.  It's too confusing, and the 'pref' bit is an implementation detail.

Comment (lbwexler) [inline core/XH.js:331]: Think we should deprecate or remove `setDarkTheme()` and make "setTheme()" (without the suffix) the main entry point.  It's too confusing, and the 'pref' bit is an implementation detail.

File: appcontainer/ThemeModel.js (hunk 2 above)
Comment (lbwexler) [reply appcontainer/ThemeModel.js:40]: also should throw if presented with a unknown value

Comment (lbwexler) [reply appcontainer/ThemeModel.js:40]: also should throw if presented with a unknown value

**Pull request merged**